### PR TITLE
Implement next episodes resolver

### PR DESCRIPTION
**Pull request by fredrikved**
Episode.next will return an array which currently only contains one episode or none.

### CODE REVIEW

File: backend/graph/api/schema/episodes.graphqls [hunk 1]
```diff
@@ -46,6 +46,8 @@ type Episode {
     lessons(first: Int, offset: Int): LessonPagination! @goField(forceResolver: true)
     shareRestriction: ShareRestriction! @goField(forceResolver: true)
     inMyList: Boolean! @goField(forceResolver: true)
+
+    next: [Episode!]! @goField(forceResolver: true)
```
Comment (andreasgangso) [inline backend/graph/api/schema/episodes.graphqls:50]: should be nullable, we may want to stop at the end of a show or playlist, for example.

File: backend/graph/api/episode-resolver.utils.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		progress, err := r.ProfileLoaders(ctx).ProgressLoader.Get(ctx, utils.AsInt(episodeId))
+		if err != nil {
+			return episodeContext, err
+		}
+		if progress != nil {
+			episodeContext = progress.Context
+		}
+	}
+
+	return episodeContext, nil
```
Comment (KillerX) [inline backend/graph/api/episode-resolver.utils.go:28]: Personally I like to invert this, i.e.:
```
if { return }
...
```

File: backend/graph/api/episode-resolver.utils.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+func (r *episodeResolver) getNextEpisodes(ctx context.Context, episodeId string) ([]int, error) {
+	episodeContext, err := r.getEpisodeContext(ctx, episodeId)
+	if err != nil {
+		return nil, err
+	}
+
+	if episodeContext.CollectionID.Valid {
```
Comment (KillerX) [inline backend/graph/api/episode-resolver.utils.go:37]: Maybe a quick comment summarizing what's going on here?

**Review by KillerX — APPROVED** [review]
Just some stylistic comments, but looks good for me otherwise.

File: backend/graph/api/schema/episodes.graphqls (hunk 1 above)
Comment (fredrikved) [reply backend/graph/api/schema/episodes.graphqls:50]: It will be empty if there's no next episodes.

Comment (andreasgangso) [reply backend/graph/api/schema/episodes.graphqls:50]: ofc ok

**Comment by fredrikved** [comment]
No, returns only 1 for now.
If we want the user to select between different episodes later, then the array will be larger. For now, there will only ever be 1 or 0 episodes in the array.
@andreasgangso

**Pull request merged**